skglm/penalties/block_separable.py

```diff
@@ -156,6 +156,98 @@ def is_penalized(self, n_features):
         return np.ones(n_features, bool_)
 
 
+spec_BlockSCAD = [
+    ('alpha', float64),
+    ('gamma', float64),
+]
+
+
+@jitclass(spec_BlockSCAD)
+class BlockSCAD(BasePenalty):
+    """Block Smoothly Clipped Absolute Deviation.
+
+    Notes
+    -----
+    With W_j the j-th row of W, the penalty is:
+        pen(||W_j||) = alpha * ||W_j||               if ||W_j|| =< alpha
+                       (2 * gamma * alpha * ||W_j|| - ||W_j|| ** 2 - alpha ** 2) \
+                           / (2 * (gamma - 1))       if alpha < ||W_j|| < alpha * gamma
+                       (alpha **2 * (gamma + 1)) / 2 if ||W_j|| > gamma * alpha
+        value = sum_{j=1}^{n_features} pen(||W_j||)
+    """
+
+    def __init__(self, alpha, gamma):
+        self.alpha = alpha
+        self.gamma = gamma
+
+    def value(self, W):
+        """Compute the value of the SCAD penalty at W."""
+        n_features = W.shape[0]
+        norm_rows = np.sqrt(np.sum(W ** 2, axis=1))
+        value = np.full_like(norm_rows, (self.gamma + 1) * self.alpha ** 2 / 2.)
+        for j in range(n_features):
+            if norm_rows[j] <= self.alpha:
+                value[j] = self.alpha * norm_rows[j]
+            elif norm_rows[j] <= self.alpha * self.gamma:
+                value[j] = (
+                    2 * self.gamma * self.alpha * norm_rows[j] - norm_rows[j] ** 2
+                    - self.alpha ** 2) / (2 * (self.gamma - 1))
+        return np.sum(value)
+
+    def prox_1feat(self, value, stepsize, j):
+        """Compute the proximal operator of BlockSCAD."""
+        nrm_value = norm(value)
```

Comment:
The formula here is the 1D SCAD prox applied to norm(value), times value / norm(value)

So we can factorize code bu putting a function prox_scad in utils, and use it both here and in single task scad.

Reply:
I have done it for SCAD and the MCP